File: Python/8kyu/8kyu-filter-out-the-geese.py
# Filter out the geese
# Language: Python
# Version: 3.10
# Difficulty: 8 kyu

# Solution 1
def goose_filter(birds):
    geese = ["African", "Roman Tufted", "Toulouse", "Pilgrim", "Steinbacher"]
    return [bird for bird in birds if not bird in geese]

# Solution 2
def goose_filter(birds):
    geese = ["African", "Roman Tufted", "Toulouse", "Pilgrim", "Steinbacher"]
    return list(filter(lambda goose: goose not in geese, birds))


# Tests
import codewars_test as test
from solution import goose_filter

@test.describe("Fixed Tests")
def fixed_tests():
    @test.it('Basic Test Cases')
    def basic_test_cases():
        test.assert_equals(goose_filter(["Mallard", "Hook Bill", "African", "Crested", "Pilgrim", "Toulouse", "Blue Swedish"]),["Mallard", "Hook Bill", "Crested", "Blue Swedish"])
        test.assert_equals(goose_filter(["Mallard", "Barbary", "Hook Bill", "Blue Swedish", "Crested"]),["Mallard", "Barbary", "Hook Bill", "Blue Swedish", "Crested"])
        test.assert_equals(goose_filter(["African", "Roman Tufted", "Toulouse", "Pilgrim", "Steinbacher"]),[])

File: R/8kyu/8kyu-filter-out-the-geese.r
# Filter out the geese
# Language: R
# Version: 3.4.1
# Difficulty: 8 kyu

goose_filter <- function(birds){
  geese = c("African", "Roman Tufted", "Toulouse", "Pilgrim", "Steinbacher")
  birds[!birds %in% geese]
}


# Tests
test_that('Basic tests', {
  expect_equal(goose_filter(c("Mallard", "Hook Bill", "African", "Crested", "Pilgrim", "Toulouse", "Blue Swedish")),c("Mallard", "Hook Bill", "Crested", "Blue Swedish"))
  expect_equal(goose_filter(c("Mallard", "Barbary", "Hook Bill", "Blue Swedish", "Crested")),c("Mallard", "Barbary", "Hook Bill", "Blue Swedish", "Crested"))
  expect_equal(goose_filter(c("African", "Roman Tufted", "Toulouse", "Pilgrim", "Steinbacher")),character())
})